### PR TITLE
Verify the type hints

### PR DESCRIPTION
**Pull request by YeonwooSung**
# Description

Simply verified the type hints for DBEmbeddingRow classes.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [v] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [v] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [v] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [v] New and existing unit tests pass locally with my changes
- [v] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-legacy/llama_index/legacy/vector_stores/lantern.py
Comment (nerdai) [inline llama-index-legacy/llama_index/legacy/vector_stores/lantern.py:1]: i think we're okay to not update `legacy` stuff at this point for these kinds of changes.

**Review by nerdai — COMMENTED** [review]
@YeonwooSung thanks for verifying the types here. 

Nit: are the sources perhaps that you can share in at least just in the PR message that you used for verification?

File: llama-index-legacy/llama_index/legacy/vector_stores/lantern.py
Comment (YeonwooSung) [reply llama-index-legacy/llama_index/legacy/vector_stores/lantern.py:1]: Fair enough, will close this PR then :)

Comment (nerdai) [reply llama-index-legacy/llama_index/legacy/vector_stores/lantern.py:1]: Oh I meant only in that the legacy folder shouldn't be updated. The others were good 👍. If you like please reopen the PR

**Pull request closed without merging**